pallets/xcm-transactor/src/lib.rs

```diff
@@ -554,12 +563,17 @@ pub mod pallet {
 
 		/// Returns weight of `transact_through_derivative` call.
 		fn weight_of_transact_through_derivative_multilocation(
-			asset: &MultiLocation,
+			asset: &VersionedMultiLocation,
 			index: &u16,
 			dest: &T::Transactor,
 			weight: &u64,
 			call: &[u8],
 		) -> Weight {
+			// If bad version, return 0
+			let asset = MultiLocation::try_from(asset.clone());
+			if asset.is_err() {
```

Comment:
It is better to refactor the code so that unwrap is not used. Instead use an if let Ok() and don't hesitate to move some code in a sub function to keep it readable :)

Reply:
better now?

Reply:
I also refactored a bit the weight functions

Reply:
Yes it's much better, congrats!